### PR TITLE
add loadConfig util

### PR DESCRIPTION
**Pull request by spideynr**
This util is meant to be used whenever we load the configuration object, so that we always have some default sane values instead of having to check for existance each time!

*Needs to be merged after #35*

### CODE REVIEW

**Comment by spideynr** [comment]
@jmike this PR looks ready to be merged. Please review and merge when you can.

File: server/utils/compileEmailTemplate.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
+import { readFileSync } from 'fs';
+import memoize from 'lodash/memoize';
 import flow from 'lodash/fp/flow';
 import template from 'lodash/template';
 import juice from 'juice';
 
-const compileEmailTemplate = flow(
-  juice,
-  template
+const readFile = (path) => readFileSync(path, 'utf8');
```
Comment (spideynr) [inline server/utils/compileEmailTemplate.js]: @jmike i wasn't 100% percent sure if i should write this differently (maybe with an apply?) i thought this was clean enough.

**Comment by spideynr** [comment]
Should we move the `yeep.config.sample.js` file inside the configuration loader or at least the utils folder?

File: server/index.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
+/* 
+**  @Deprecation notice
```
Comment (jmike) [inline server/index.js]: I actually wouldn't want to deprecate this - my intention is to keep it solely for development. In production we use the CLI as discussed.

File: server/utils/compileEmailTemplate.js (hunk 1 above)
Comment (jmike) [reply server/utils/compileEmailTemplate.js]: It's cool what you did here - FP all the way!

There might be a better way to do this using [partialRight()](https://lodash.com/docs/4.17.11#partialRight), e.g.

```javascript
const compileEmailTemplate = memoize(
  flow(
    partialRight(readFileSync, 'utf8'),
    juice,
    template
  )
);
```

**Comment by jmike** [comment]
@spideynr I agree with your suggestion - we should move `yeep.config.sample.js` to the same directory as `loadConfig()`. I would personally keep it in a separate file and simply change the name to `yeep.config.default.js` or something similar.

**Pull request merged**